sre_yield/__init__.py

```diff
@@ -497,9 +500,19 @@ def __init__(
         if not isinstance(pattern, sre_parse.SubPattern):
             pattern = sre_parse.parse(pattern, flags)
         self.matcher = sre_compile.compile(pattern, flags)
-        if not flags & re.DOTALL:
-            charset = "".join(c for c in charset if c != "\n")
-        self.charset = charset
+        self.categories = CATEGORIES.copy()
+        if flags & re.DOTALL:
+            self.categories[sre_constants.CATEGORY_LINEBREAK] = ""
+            self.categories[sre_constants.CATEGORY_NOT_LINEBREAK] = CHARSET
```

Comment:
I think this should be `= charset` lowercase

Reply:
Good catch.  Otherwise, does this approach seem reasonable, and worth polishing?